File: src/main/java/com/autosale/shop/repository/impl/UserRepositoryImpl.java
package com.autosale.shop.repository.impl;


import com.autosale.shop.model.User;
import com.autosale.shop.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.jooq.DSLContext;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

import static shop.domain.tables.Users.USERS;


@RequiredArgsConstructor
@Repository
public class UserRepositoryImpl implements UserRepository {

    private final DSLContext dsl;

    @Override
    public List<User> findAll() {
        return dsl.selectFrom(USERS)
                .fetchInto(User.class);
    }

    @Override
    public Optional<User> findById(int id) {
        return dsl.selectFrom(USERS)
                .where(USERS.ID.eq(id))
                .fetchOptionalInto(User.class);
    }

    @Override
    public Optional<Integer> save(User user) {
        return dsl.insertInto(USERS)
                .set(dsl.newRecord(USERS, user))
                .returningResult(USERS.ID)
                .fetchOptionalInto(Integer.class);
    }

    @Override
    public void update(User user) {
        dsl.newRecord(USERS,user)
                .update();

    }

    @Override
    public int deleteById(int id) {
        return dsl.deleteFrom(USERS)
                .where(USERS.ID.eq(id))
                .execute();
    }

}

File: src/main/java/com/autosale/shop/repository/impl/UserRepositoryImpl.kt
package com.autosale.shop.repository.impl

import com.autosale.shop.model.User
import com.autosale.shop.repository.UserRepository
import org.jooq.DSLContext
import org.springframework.stereotype.Repository
import structure.tables.Users
import java.util.*


@Repository
class UserRepositoryImpl(private val dsl: DSLContext) : UserRepository {
    override fun findAll(): List<User> {
        return dsl.selectFrom(Users.USERS)
            .fetchInto(User::class.java)
    }

    override fun findById(id: Int): Optional<User> {
        return dsl.selectFrom(Users.USERS)
            .where(Users.USERS.ID.eq(id))
            .fetchOptionalInto(User::class.java)
    }

    override fun findByUsername(userName: String): Optional<User> {
        return dsl.selectFrom(Users.USERS)
            .where(Users.USERS.USER_NAME.eq(userName))
            .fetchOptionalInto(User::class.java)
    }

    override fun save(user: User): Optional<Int> {
        return dsl.insertInto(Users.USERS)
            .set(dsl.newRecord(Users.USERS, user))
            .returningResult(Users.USERS.ID)
            .fetchOptionalInto(Int::class.java)
    }

    override fun update(user: User) {
        dsl.newRecord(Users.USERS, user)
            .update()
    }

    override fun deleteById(id: Int): Int {
        return dsl.deleteFrom(Users.USERS)
            .where(Users.USERS.ID.eq(id))
            .execute()
    }
}
